src/main/java/wolox/training/models/Book.java

```diff
@@ -62,6 +63,9 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        Preconditions.checkNotNull(genre);
```

Comment:
When checking both things (usually on Strings), you could simply write:
Preconditions.checkArgument(genre != Null && !genre.isEmpty);